### ISSUE TITLE
3X+ speed regression with sphinx 2.4 over 2.3

### ISSUE THREAD

**Issue by kapilt** [sphinx-doc/sphinx#7343]
**Describe the bug**
I noticed our doc builds getting considerably slower after upgrading to latest sphinx (toctree cached and -j) from our ci doc builds, while poking around (-vvv and disable -j, output sample below) i noticed that a significant amount of time being spent in sphinx translator visiting nodes, and it looks like the issue is related to this blurb in the changelog

`SphinxTranslator now calls visitor/departure method for super node class if visitor/departure method for original node class not found`

our builds in ci (azure pipelines) went from 5m to 18m after the upgrade. downgrading to 2.3 restores speed to ~5m.

```
SphinxTranslator.dispatch_departure calling depart_Text for Filters
SphinxTranslator.dispatch_departure calling depart_reference for <reference anchorname="#filters" internal="True" refuri="../resources/rds-cluster-param-group.html#filters">Filters</reference>                                                                                                                                                                                            
SphinxTranslator.dispatch_departure calling depart_compact_paragraph for <compact_paragraph classes="toctree-l4"><reference anchorname="#filters" internal="True" refuri="../resources/rds-cluster-param-group.html#filters">Filters</reference></compact_paragraph>                                                                                                                        
SphinxTranslator.dispatch_departure calling depart_list_item for <list_item classes="toctree-l4"><compact_paragraph classes="toctree-l4"><reference anchorname="#filters" internal="True" refuri="../resources/rds-cluster-param-group.html#filters">Filters</reference></compact_paragraph></list_item>                                                                                    
SphinxTranslator.dispatch_visit calling visit_list_item for <list_item classes="toctree-l4"><compact_paragraph classes="toctree-l4"><reference anchorname="#actions" internal="True" refuri="../resources/rds-cluster-param-group.html#actions">Actions</reference></compact_paragraph></list_item>                                                                                         
SphinxTranslator.dispatch_visit calling visit_compact_paragraph for <compact_paragraph classes="toctree-l4"><reference anchorname="#actions" internal="True" refuri="../resources/rds-cluster-param-group.html#actions">Actions</reference></compact_paragraph>                                                                                                                             
SphinxTranslator.dispatch_visit calling visit_reference for <reference anchorname="#actions" internal="True" refuri="../resources/rds-cluster-param-group.html#actions">Actions</reference>
SphinxTranslator.dispatch_visit calling visit_Text for Actions
SphinxTranslator.dispatch_departure calling depart_Text for Actions
SphinxTranslator.dispatch_departure calling depart_reference for <reference anchorname="#actions" internal="True" refuri="../resources/rds-cluster-param-group.html#actions">Actions</reference>                                                                                                                                                                                            
SphinxTranslator.dispatch_departure calling depart_compact_paragraph for <compact_paragraph classes="toctree-l4"><reference anchorname="#actions" internal="True" refuri="../resources/rds-cluster-param-group.html#actions">Actions</reference></compact_paragraph>                                                                                                                        
SphinxTranslator.dispatch_departure calling depart_list_item for <list_item classes="toctree-l4"><compact_paragraph classes="toctree-l4"><reference anchorname="#actions" internal="True" refuri="../resources/rds-cluster-param-group.html#actions">Actions</reference></compact_paragraph></list_item>                                                                                    
SphinxTranslator.dispatch_departure calling depart_bulle
```

**Expected behavior**

**Your project**
https://github.com/cloud-custodian/cloud-custodian/tree/master/docs

**Environment info**
- Python version: 3.8.
- Sphinx version: 2.4.4
- Sphinx extensions:  
sphinx==2.4.4
sphinx-markdown-tables==0.0.12
sphinx-rtd-theme==0.4.3
recommonmark==0.6.0

**Comment by tk0miya** [comment]
Sorry for inconvenience. As a workaround, it will be resolved with disabling debug logs.
Thanks,

**Comment by tk0miya** [comment]
Fixed by #7344 .
Thank you for reporting!

**Comment by kapilt** [comment]
thanks for the fast fix :-)
